### PR TITLE
Add an easy way to reobfuscate multiple jars.

### PR DESCRIPTION
**Pull request by killjoy1221**
Example usage: https://gist.github.com/killjoy1221/c9487befa9cafd65054f

### CODE REVIEW

**Comment by killjoy1221** [comment]
I've added srgExtra now.  TODO: Config defaults

**Comment by AbrarSyed** [comment]
please squash

**Comment by killjoy1221** [comment]
All squashed.

File: src/main/java/net/minecraftforge/gradle/user/ReobfConfig.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        this.classpath = classpath;
+    }
+
+    public FileCollection getClasspath()
+    {
+        return classpath;
+    }
+
+    public void setSrgMcp()
```
Comment (matthewprenger) [inline src/main/java/net/minecraftforge/gradle/user/ReobfConfig.java]: This should be called `setSrgNotch` not `setSrgMcp`.

File: src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java [hunk 2]
```diff
@@ -204,6 +220,34 @@ public void execute(TaskSourceCopy t)
...
+
+            task.setClasspath(reobf.getClasspath());
+            task.setJar(((Jar) jar).getArchivePath());
+            task.setPrimarySrg(delayedFile(reobf.getSrg()));
+
+            task.dependsOn(TASK_GENERATE_SRGS, jar);
+            task.mustRunAfter("test");
+
+            project.getTasks().getByName("build").dependsOn(task);
+            project.getTasks().getByName("assemble").dependsOn(task);
```
Comment (matthewprenger) [inline src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java]: `build` already depends on `assemble` so having both of these statements is redundant.

**Comment by AbrarSyed** [comment]
I think it would be better to explicitely force the specification of the SRG as opposed to using the plugin default (srg names for forge).
Should also allow for secondary SRG files, and extra SRG lines. It might actually be a good idea to have a common interface for the reobf task and this reobf config object, or something that can just be copy-pasted between them...

**Comment by killjoy1221** [comment]
Oops. I must've accidentally removed the srg extra when I squashed. I'll have to fix that. 

I chose to make mcp the default because only Forge used the srg names, which means I only had to override the setupReobf method in the Forge plugin.

**Comment by AbrarSyed** [comment]
Also, please add javadocs to your new reobf extension thing. When I eventually get arround to writing the docs, I will be frequently linking to the javadocs for more information about specific features. It would also be nice to add some javadocs to the setupReobf method just to remind me what it does.. 

I really need to javadoc more...

File: src/main/java/net/minecraftforge/gradle/user/TaskSingleReobf.java [hunk 3]
```diff
@@ -268,6 +268,61 @@ public File getPrimarySrg()
...
+        list.addAll(extraSrgLines);
+        return list;
+    }
+
+    @Override
+    public void extra(Collection<Object> o)
+    {
+        for (Object obj : o)
+        {
+            if (obj instanceof String)
```
Comment (matthewprenger) [inline src/main/java/net/minecraftforge/gradle/user/TaskSingleReobf.java]: Might need a `GString` check here too.

**Comment by AbrarSyed** [comment]
I wanna release 2.0.2 tonight, please finish this ASAP.

**Comment by killjoy1221** [comment]
All done. ready for pull.

**Comment by killjoy1221** [comment]
Got everything working for real this time. Just need to squash.

**Pull request merged**